text/0000-pack-subcommands.md

```diff
@@ -0,0 +1,206 @@
+  set-default-builder   Set default builder used by other commands
+  inspect-builder       Show information about a builder
+  suggest-builders      Display list of recommended builders
+  trust-builder         Trust builder
+  untrust-builder       Stop trusting builder
+  list-trusted-builders List Trusted Builders
+  create-builder        Create builder image
+  package-buildpack     Package buildpack in OCI format.
+  suggest-stacks        Display list of recommended stacks
+  version               Show current 'pack' version
+  report                Display useful information for reporting an issue
+  completion            Outputs completion script location
+  help                  Help about any command
+```
+
+# What it is
+[what-it-is]: #what-it-is
+
+
+### Proposed structure
+
+The pack commands will be reorganized in a way where they are grouped by resource type where possible. Additionally, a new
+command `config` will group subcommands that solely operate on mutating the config file.
+
+
+An overview of the (sub)commands:
+
+```
+pack
+
+  build
+  builder
+    create
+    inspect
+    suggest
+  buildpack
+    package
+    register
+  completion
+  config
+    default-builder
+      set
+      unset
+      show
+    run-image-mirror
+      add
+      remove
+      list
+    registry
+      add
+      remove
+      list
+    trusted-builder
+      add
+      remove
+      list
+  help
+  inspect
+  report
+  rebase
+  stack
+    suggest
+  version
+```
+
+### Config
+
+The `config` subcommand would work with the notion of two different type of resources:
+
+##### Singular
+
+Singular resources would have the following subcommands:
+
+* set
+* unset
+* show (aliased to subcommand root)
+
+###### Example
+
+```
+# sets the default builder
+pack config default-builder set gcr.io/my-org/my-builder:latest
+
+# unset the default builder
+pack config default-builder unset
+
+# show the default builder
+pack config default-builder show
+
+# show the default builder (w/o subcommand)
+pack config default-builder
+```
+
+##### List
+
+List resources would have the following subcommands:
+
+* add
+* remove
+* list
+
+The explicit additional parameters would be based on the specific implementation needs.
+
+###### Example
+
+```
+# adds `gcr.io/my-org/my-image` as a mirror for `my-image`  
+pack config run-image-mirrors add my-image gcr.io/my-org/my-image
+
+# removes `gcr.io/my-org/my-image` as a mirror for `my-image`
+pack config run-image-mirrors remove my-image gcr.io/my-org/my-image
+
+# removes all mirrors for `my-image`
+pack config run-image-mirrors remove my-image
+
+# list all mirrors
+pack config run-image-mirrors list
+
+# list all mirrors for `my-image`
+pack config run-image-mirrors list my-image
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+
+## Backwards Compatibility
+
+1. There would be a period in which all pre-existing subcommands would work.
```

Comment:
Maybe it's worth adding that a warning should appear once running the pre-existing subcommands.

Reply:
```suggestion
1. There would be a period in which all pre-existing subcommands would work (with a deprecation warning).
```